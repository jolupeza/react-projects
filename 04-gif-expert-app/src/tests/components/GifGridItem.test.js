import { shallow } from 'enzyme';
import GifGridItem from '../../components/GiftGridItem';

describe('Testing component GifGridItem', () => {
  const title = 'My title';
  const url = 'https://test.net/img.jpg';

  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show component GifGridItem', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a parragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have a image equal to url and attribute alt', () => {
    const image = wrapper.find('img');
    expect(image.prop('src')).toBe(url);
    expect(image.prop('alt')).toBe(title);
  });

  test('should have class animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animated__fadeIn')).toBe(true);
  });
});
