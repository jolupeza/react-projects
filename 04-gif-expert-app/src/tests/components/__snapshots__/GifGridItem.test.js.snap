// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component GifGridItem should show component GifGridItem 1`] = `
<div
  className="card animated__fadeIn"
>
  <img
    alt="My title"
    src="https://test.net/img.jpg"
  />
  <p>
    My title
  </p>
</div>
`;
